gpt4all-backend/CMakeLists.txt

```diff
@@ -9,7 +9,9 @@ if(APPLE)
     set(CMAKE_OSX_ARCHITECTURES "arm64;x86_64" CACHE STRING "" FORCE)
   else()
     # Build for the host architecture on macOS
-    set(CMAKE_OSX_ARCHITECTURES "${CMAKE_HOST_SYSTEM_PROCESSOR}" CACHE STRING "" FORCE)
+    if(NOT CMAKE_OSX_ARCHITECTURES)
+      set(CMAKE_OSX_ARCHITECTURES "${CMAKE_HOST_SYSTEM_PROCESSOR}" CACHE STRING "" FORCE)
+    endif()
```

Comment:
This is required to allow for cross-compiling macOS, iOS, and Catalyst libraries. If you disable UNIVERSAL, it would use whatever your hosts CPU was for building, which doesn't work if you want to build Mac Catalyst libraries (where you need to build them for separate archs and `lipo` them) or iOS Device builds if you're on an Intel CPU (since that needs to be ARM64, and if you set it here on Intel, Xcode will cross-compile the library)

Reply:
The change does not affect normal GPT4All Builds in the repo. All it is saying is that if you had already set "CMAKE_OSX_ARCHITECTURES" (Like in scripts I have written to enable Mac Catalyst builds) it will use those instead. Else, it will set it to use your host system values.

If you're building Gpt4all from within the repo, it doesn't change anything, the defaults are the same.